ffmpeg/extras.c

```diff
@@ -176,6 +208,7 @@ int lpms_get_codec_info(char *fname, pcodec_info out)
       }
       out->width  = ic->streams[vstream]->codecpar->width;
       out->height = ic->streams[vstream]->codecpar->height;
+      out->fps = av_q2d(ic->streams[vstream]->avg_frame_rate);
```

Comment:
Note that [`avg_frame_rate`](https://ffmpeg.org/doxygen/trunk/structAVStream.html#a946e1e9b89eeeae4cab8a833b482c1ad)  is not guaranteed to be set to anything. If unset, [`r_frame_rate`](https://ffmpeg.org/doxygen/trunk/structAVStream.html#ad63fb11cc1415e278e09ddc676e8a1ad) might be acceptable but I am not sure how this value would be used.

Reply:
r_frame_rate is working well, seems more reliable

Reply:
note that `r_frame_rate` does not interact well with VFR or otherwise unreliable streams (RTP / webrtc) - this caused a big issue in the past, see #393

the suggestion was more to use `r_frame_rate` if `avg_frame_rate` is unset, but I am not sure which one you want to trust more

ffmpeg keeps them separate for a reason - so the caller can distinguish between what's declared in the container, vis-a-vis a calculated approximation from the actual data